ratelimit/keeper/hooks.go

```diff
@@ -25,22 +21,23 @@ func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochInfo epochstypes.EpochInf
 	}
 }
 
-func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochInfo epochstypes.EpochInfo) {}
-
-type Hooks struct {
-	k Keeper
-}
-
-var _ epochstypes.EpochHooks = Hooks{}
-
-func (k Keeper) Hooks() Hooks {
-	return Hooks{k}
-}
-
-func (h Hooks) BeforeEpochStart(ctx sdk.Context, epochInfo epochstypes.EpochInfo) {
-	h.k.BeforeEpochStart(ctx, epochInfo)
-}
+// Checks if it's time to start the new hour epoch
+func (k Keeper) CheckHourEpochStarting(ctx sdk.Context) (epochStarting bool, epochNumber uint64) {
+	hourEpoch := k.GetHourEpoch(ctx)
+
+	// If the block time is later than the current epoch start time + epoch duration,
+	// move onto the next epoch by incrementing the epoch number, height, and start time
+	currentEpochEndTime := hourEpoch.EpochStartTime.Add(hourEpoch.Duration)
```

Comment:
Probably it would be cleaner to have the epoch duration a param of the `ratelimit` module. I guess that with the current implementation, you need a software upgrade to update the duration, right? If so, why not just hardcode it as a constant and remove it from the `HourEpoch` struct?

Reply:
My thinking was that in practice you would only ever have this set to a "true hour" on mainnet. Rate limit windows are denominated in hours, so the hour epoch should always be exactly 1 hour and then you change the length of the rate limit window by changing the number of hours

And the only time it would deviate from this is in testing - in which case you really just need to be able to set it to something shorter in the genesis.json.

---

That said, I think params is an equally valid place for it - it just seemed slightly more intuitive to put it on the epoch struct where it's referenced. 

Wdyt? Happy to make the change either way

Reply:
I think it's fine like this for now. Also, changing it and adding params might require state migration on Stride, which is not ideal.